src/pages/EditRequestPage.js

```diff
@@ -171,6 +178,23 @@ function EditRequestPage({report, route, parentReport, policy, session}) {
         );
     }
 
+    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.CATEGORY) {
+        return (
+            <EditRequestCategoryPage
+                defaultCategory={transactionCategory}
+                policyID={lodashGet(report, 'policyID', '')}
+                onSubmit={(transactionChanges) => {
+                    let updatedCategory = transactionChanges.category;
+                    // In case the same category has been selected, do reset of the category.
+                    if (transactionCategory === updatedCategory) {
+                        updatedCategory = '';
+                    }
+                    editMoneyRequest({category: updatedCategory});
+                }}
```

Comment:
Same regarding to extraction

Reply:
I did it like an inline-callback to follow an approach for this file that we have before. So, I guess we can improve it during some refactoring issues like TS migration, class to func. and so on.

<img width="677" alt="Screenshot 2023-09-19 at 11 35 04" src="https://github.com/Expensify/App/assets/57314631/c2fdec6c-728b-45dd-9b77-a660a7ca177f">